.ci/ci.yml

```diff
@@ -133,6 +133,11 @@ jobs:
     ccacheArtifactName: 'ccache-macos'
     ccacheArchive: '$(Build.ArtifactStagingDirectory)/ccache-$(Build.BuildId).tar'
     commonMacOSCMakeFlags: '-DBUILD_EXAMPLES=OFF -DBUNDLE_JSON=OFF -DBUNDLE_NLOPT=OFF -DENABLE_TESTING=ON -DENABLE_COVERAGE=OFF -DBUILD_META_EXAMPLES=OFF'
+    openMPMacOSCMakeFlags:
+      '-DOpenMP_CXX_FLAGS="-Xpreprocessor -fopenmp"
+      -DOpenMP_C_FLAGS="-Xpreprocessor -fopenmp"
+      -DOpenMP_CXX_LIB_NAMES="omp" -DOpenMP_C_LIB_NAMES="omp"
+      -DOpenMP_omp_LIBRARY="/usr/local/opt/libomp/lib/libomp.dylib"'
```

Comment:
i wonder whether we can detect this rather than using this fixed path?

Reply:
We could use `$(brew --prefix libomp)/lib/libomp.dylib`.

Reply:
Another option would be to use llvm clang instead of xcode clang. New versions of llvm clang come with openmp built in, so I believe in that case we wouldn't need the openmp flags at all, and cmake would detect it.

Reply:
imo that would be much better

Reply:
the brew stuff or clang stuff?
+ sorry, my mistake, I actually used `$(brew --prefix libomp)` when testing.